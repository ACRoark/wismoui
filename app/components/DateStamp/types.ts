export type DateStampFormat = 'auto' | 'long' | 'short';
