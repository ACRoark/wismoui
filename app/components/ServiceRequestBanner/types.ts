export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
