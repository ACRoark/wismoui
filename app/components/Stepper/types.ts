import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
