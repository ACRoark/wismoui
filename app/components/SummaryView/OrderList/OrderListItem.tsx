import React from 'react';
import { FormattedMessage } from 'react-intl';

import Address from 'components/Address';
import Hyperlink from 'components/Hyperlink';
import OrderNumber from 'components/OrderNumber';
import { IOrderRequestInfo } from 'types';

import NewOrderListItemTitle from './NewOrderListItemTitle';
import OrdersListItemIcon from './OrderListItemIcon';

import messages from './messages';

interface IOrderListItemProps {
  orderNumber: string;
  request: IOrderRequestInfo;
}

const ChevronRight = () => (
  <svg
    className="bi bi-chevron-right"
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M4.646 1.646a.5.5 0 01.708 0l6 6a.5.5 0 010 .708l-6 6a.5.5 0 01-.708-.708L10.293 8 4.646 2.354a.5.5 0 010-.708z"
      fillRule="evenodd"
      stroke="currentColor"
      strokeWidth="1"
    />
  </svg>
);

const OrderListItem: React.FC<IOrderListItemProps> = (props: IOrderListItemProps): React.ReactElement => {
  const { orderNumber, request } = props;

  const { address, orderRequestType } = request;

  return (
    <div className="dte-wismo-order-list-item" id={`order-${orderNumber}`}>
      <div className="icon">
        <OrdersListItemIcon requestType={orderRequestType}/>
      </div>
      <div className="content">
        <div className="content-header">
          <NewOrderListItemTitle request={request}/>
          <div className="order-number">
            <span className="label">
              <FormattedMessage {...messages.orderNumber} />
            </span>
            <span className="value">
              <OrderNumber value={orderNumber}/>
            </span>
          </div>
        </div>
        <div className="address">
          <Address address={address}/>
        </div>
        <div className="details-link">
          <Hyperlink href={`/orders/${orderNumber}`}>
            <FormattedMessage {...messages.detailsLinkText} />
            <ChevronRight/>
          </Hyperlink>
        </div>
      </div>
    </div>
  );
};

export default OrderListItem;
