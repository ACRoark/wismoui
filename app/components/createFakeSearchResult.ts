import * as _ from 'lodash';
import { IOrderRequest, IOrderSearchResult } from 'types';
import IOrderRequestInfo from 'types/IOrderRequestInfo';

import sortOrderRequestUpdates from 'utils/sortOrderRequestUpdates';
import createFakeOrder from './createFakeOrder';

const createFakeSearchResult = (orderNumber: string): IOrderSearchResult => {
  const order = createFakeOrder(orderNumber);

  return {
    closedDetails: order.closedDetails,
    createdAt: order.createdAt,
    createdBy: order.createdBy,
    customer: order.customer,
    orderNumber,
    orderRequests: order.orderRequests.map(
      (request: IOrderRequest): IOrderRequestInfo => ({
        accountNumber: request.accountNumber,
        address: request.address,
        currentOrderRequestStatus: _.last(sortOrderRequestUpdates(request.orderRequestStatusUpdates)),
        orderRequestType: request.orderRequestType,
        premiseId: request.premiseId,
        products: request.products,
        verificationEvents: [],
        wantDate: request.wantDate,
      }),
    ),
    orderType: order.orderType,
  };
};

export default createFakeSearchResult;
