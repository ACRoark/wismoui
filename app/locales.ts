export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
