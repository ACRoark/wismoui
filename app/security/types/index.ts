import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
