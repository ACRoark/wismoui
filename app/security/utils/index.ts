import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
