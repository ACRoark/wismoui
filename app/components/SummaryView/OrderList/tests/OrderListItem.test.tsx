import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import { SUPPORTED_LOCALES } from 'locales';

import createFakeSearchResult from 'components/createFakeSearchResult';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { OrderRequestType, OrderRequestTypes } from 'types';

import OrderListItem from '../OrderListItem';

// NOTE: We cannot use generate.OrderRequest because we need static data for snapshots;
//       otherwise, the snapshot would be different each time it is run.
const orderNumber = 'MI12345678';
const baseRequest = createFakeSearchResult(orderNumber).orderRequests[0];
const store = createMockStore();

const expectNoConsoleErrors = (language: string, requestType: OrderRequestType): void => {
  const spy = jest.spyOn(global.console, 'error');

  const request = {
    ...baseRequest,
    OrderRequestType: requestType,
  };

  renderWithIntl(
    <Router>
      <Provider store={store}>
        <OrderListItem orderNumber={orderNumber} request={request} />
      </Provider>
    </Router>, language);

  expect(spy).not.toHaveBeenCalled();
};

const expectComponentToMatchSnapshot = (language: string, requestType: OrderRequestType): void => {
  const request = {
    ...baseRequest,
    OrderRequestType: requestType,
  };

  const tree = createSnapshotWithIntl(
    <Router>
      <Provider store={store}>
        <OrderListItem orderNumber={orderNumber} request={request} />
      </Provider>
    </Router>, language);

  expect(tree).toMatchSnapshot();
};

describe('<OrderListItem />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    OrderRequestTypes.forEach((requestType: OrderRequestType): void => {
      it(`should not log errors in console when rendered for '${requestType}' requests and the current language is ${language}`, (): void =>
        expectNoConsoleErrors(language, requestType));

      it(`should render correctly when the request type is '${requestType}' and the current language is ${language}`, (): void =>
        expectComponentToMatchSnapshot(language, requestType));
    });
  });
});
