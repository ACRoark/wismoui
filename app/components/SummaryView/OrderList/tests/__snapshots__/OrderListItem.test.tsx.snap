// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderListItem /> should render correctly when the request type is 'MIMO_START' and the current language is en 1`] = `
<div
  className="dte-wismo-order-list-item"
  id="order-MI12345678"
>
  <div
    className="icon"
  >
    <svg
      className="dte-wismo-order-list-item-icon"
      fill="none"
      height="58"
      viewBox="0 0 34 58"
      width="34"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        start order
      </title>
      <path
        d="M1 16C1 15.4477 1.44772 15 2 15H32C32.5523 15 33 15.4477 33 16V37C33 40.866 29.866 44 26 44H8C4.13401 44 1 40.866 1 37V16Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M7 3C7 1.89543 7.89543 1 9 1H10C11.1046 1 12 1.89543 12 3V15H7V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M22 3C22 1.89543 22.8954 1 24 1H25C26.1046 1 27 1.89543 27 3V15H22V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M5.02124 56.3499C7.73517 56.3499 13.1835 56.6666 15.4086 53.9259C17.6338 51.1852 18.1999 45.985 17.6012 44.0577"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M18.8224 39.0071L14.5581 29.6487L20.4274 29.7357L16.1777 19.3992"
        stroke="currentColor"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <div
      className="content-header"
    >
      <div
        className="title"
      >
        Start Service 
      </div>
      <div
        className="order-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span
          className="value"
        >
          <span>
            MI 1234 5678
          </span>
        </span>
      </div>
    </div>
    <div
      className="address"
    >
      <address
        className="dte-wismo-address"
      >
        1 Bagshot Row
Hobbiton, TS 02890
      </address>
    </div>
    <div
      className="details-link"
    >
      <a
        href="/orders/MI12345678"
        onClick={[Function]}
        target="_self"
      >
        Details
        <svg
          className="bi bi-chevron-right"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.646 1.646a.5.5 0 01.708 0l6 6a.5.5 0 010 .708l-6 6a.5.5 0 01-.708-.708L10.293 8 4.646 2.354a.5.5 0 010-.708z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`<OrderListItem /> should render correctly when the request type is 'MIMO_START' and the current language is es 1`] = `
<div
  className="dte-wismo-order-list-item"
  id="order-MI12345678"
>
  <div
    className="icon"
  >
    <svg
      className="dte-wismo-order-list-item-icon"
      fill="none"
      height="58"
      viewBox="0 0 34 58"
      width="34"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        start order
      </title>
      <path
        d="M1 16C1 15.4477 1.44772 15 2 15H32C32.5523 15 33 15.4477 33 16V37C33 40.866 29.866 44 26 44H8C4.13401 44 1 40.866 1 37V16Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M7 3C7 1.89543 7.89543 1 9 1H10C11.1046 1 12 1.89543 12 3V15H7V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M22 3C22 1.89543 22.8954 1 24 1H25C26.1046 1 27 1.89543 27 3V15H22V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M5.02124 56.3499C7.73517 56.3499 13.1835 56.6666 15.4086 53.9259C17.6338 51.1852 18.1999 45.985 17.6012 44.0577"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M18.8224 39.0071L14.5581 29.6487L20.4274 29.7357L16.1777 19.3992"
        stroke="currentColor"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <div
      className="content-header"
    >
      <div
        className="title"
      >
        Empieza el servicio 
      </div>
      <div
        className="order-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span
          className="value"
        >
          <span>
            MI 1234 5678
          </span>
        </span>
      </div>
    </div>
    <div
      className="address"
    >
      <address
        className="dte-wismo-address"
      >
        1 Bagshot Row
Hobbiton, TS 02890
      </address>
    </div>
    <div
      className="details-link"
    >
      <a
        href="/orders/MI12345678"
        onClick={[Function]}
        target="_self"
      >
        Detalles
        <svg
          className="bi bi-chevron-right"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.646 1.646a.5.5 0 01.708 0l6 6a.5.5 0 010 .708l-6 6a.5.5 0 01-.708-.708L10.293 8 4.646 2.354a.5.5 0 010-.708z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`<OrderListItem /> should render correctly when the request type is 'MIMO_STOP' and the current language is en 1`] = `
<div
  className="dte-wismo-order-list-item"
  id="order-MI12345678"
>
  <div
    className="icon"
  >
    <svg
      className="dte-wismo-order-list-item-icon"
      fill="none"
      height="58"
      viewBox="0 0 34 58"
      width="34"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        start order
      </title>
      <path
        d="M1 16C1 15.4477 1.44772 15 2 15H32C32.5523 15 33 15.4477 33 16V37C33 40.866 29.866 44 26 44H8C4.13401 44 1 40.866 1 37V16Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M7 3C7 1.89543 7.89543 1 9 1H10C11.1046 1 12 1.89543 12 3V15H7V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M22 3C22 1.89543 22.8954 1 24 1H25C26.1046 1 27 1.89543 27 3V15H22V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M5.02124 56.3499C7.73517 56.3499 13.1835 56.6666 15.4086 53.9259C17.6338 51.1852 18.1999 45.985 17.6012 44.0577"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M18.8224 39.0071L14.5581 29.6487L20.4274 29.7357L16.1777 19.3992"
        stroke="currentColor"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <div
      className="content-header"
    >
      <div
        className="title"
      >
        Start Service 
      </div>
      <div
        className="order-number"
      >
        <span
          className="label"
        >
          Order No.
        </span>
        <span
          className="value"
        >
          <span>
            MI 1234 5678
          </span>
        </span>
      </div>
    </div>
    <div
      className="address"
    >
      <address
        className="dte-wismo-address"
      >
        1 Bagshot Row
Hobbiton, TS 02890
      </address>
    </div>
    <div
      className="details-link"
    >
      <a
        href="/orders/MI12345678"
        onClick={[Function]}
        target="_self"
      >
        Details
        <svg
          className="bi bi-chevron-right"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.646 1.646a.5.5 0 01.708 0l6 6a.5.5 0 010 .708l-6 6a.5.5 0 01-.708-.708L10.293 8 4.646 2.354a.5.5 0 010-.708z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`<OrderListItem /> should render correctly when the request type is 'MIMO_STOP' and the current language is es 1`] = `
<div
  className="dte-wismo-order-list-item"
  id="order-MI12345678"
>
  <div
    className="icon"
  >
    <svg
      className="dte-wismo-order-list-item-icon"
      fill="none"
      height="58"
      viewBox="0 0 34 58"
      width="34"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        start order
      </title>
      <path
        d="M1 16C1 15.4477 1.44772 15 2 15H32C32.5523 15 33 15.4477 33 16V37C33 40.866 29.866 44 26 44H8C4.13401 44 1 40.866 1 37V16Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M7 3C7 1.89543 7.89543 1 9 1H10C11.1046 1 12 1.89543 12 3V15H7V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M22 3C22 1.89543 22.8954 1 24 1H25C26.1046 1 27 1.89543 27 3V15H22V3Z"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M5.02124 56.3499C7.73517 56.3499 13.1835 56.6666 15.4086 53.9259C17.6338 51.1852 18.1999 45.985 17.6012 44.0577"
        stroke="currentColor"
        strokeWidth="2"
      />
      <path
        d="M18.8224 39.0071L14.5581 29.6487L20.4274 29.7357L16.1777 19.3992"
        stroke="currentColor"
        strokeWidth="2"
      />
    </svg>
  </div>
  <div
    className="content"
  >
    <div
      className="content-header"
    >
      <div
        className="title"
      >
        Empieza el servicio 
      </div>
      <div
        className="order-number"
      >
        <span
          className="label"
        >
          No. de pedido
        </span>
        <span
          className="value"
        >
          <span>
            MI 1234 5678
          </span>
        </span>
      </div>
    </div>
    <div
      className="address"
    >
      <address
        className="dte-wismo-address"
      >
        1 Bagshot Row
Hobbiton, TS 02890
      </address>
    </div>
    <div
      className="details-link"
    >
      <a
        href="/orders/MI12345678"
        onClick={[Function]}
        target="_self"
      >
        Detalles
        <svg
          className="bi bi-chevron-right"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.646 1.646a.5.5 0 01.708 0l6 6a.5.5 0 010 .708l-6 6a.5.5 0 01-.708-.708L10.293 8 4.646 2.354a.5.5 0 010-.708z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
